src/TLuaInterpreter.cpp

```diff
@@ -17659,3 +17662,118 @@ int TLuaInterpreter::showNotification(lua_State* L)
     return 0;
 }
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#registerMapInfoProvider
+int TLuaInterpreter::registerMapInfoProvider(lua_State* L)
+{
+    auto& host = getHostFromLua(L);
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "registerMapInfoProvider: bad argument #1 type (label as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    if (!lua_isfunction(L, 2)) {
+        lua_pushfstring(L, "registerMapInfoProvider: bad argument #2 type (callback as function expected, got %s!)", luaL_typename(L, 2));
+        return lua_error(L);
+    }
+
+    auto name = lua_tostring(L, 1);
+    int callback = luaL_ref(L, LUA_REGISTRYINDEX);
+
+    host.mpMap->mpMapper->mMapInfoContributorManager->contributors.insert(name, [=](int roomID, int selectionSize, int areaId, bool showingCurrentArea, QColor& infoColor) {
+        lua_rawgeti(L, LUA_REGISTRYINDEX, callback);
+        if (roomID > 0) {
+            lua_pushinteger(L, roomID);
+        } else {
+            lua_pushnil(L);
+        }
+        lua_pushinteger(L, selectionSize);
+        lua_pushinteger(L, areaId);
+        lua_pushboolean(L, showingCurrentArea);
+        
+        lua_pcall(L, 4, 6, 0);
+
+        auto nResult = lua_gettop(L);
+        auto index = -nResult;
+        QString text;
+        text = lua_tostring(L, index);
+        bool isBold = lua_toboolean(L, ++index);
+        bool isItalic = lua_toboolean(L, ++index);
+        int r = -1;
+        int g = -1;
+        int b = -1;
+        if(!lua_isnil(L, ++index)) {
+            r = lua_tonumber(L, index);
+        }
+        if(!lua_isnil(L, ++index)) {
+            g = lua_tonumber(L, index);
+        }
+        if(!lua_isnil(L, ++index)) {
+            b = lua_tonumber(L, index);
+        }
+        QColor color;
+        if (r >= 0 && r <= 255 && b >= 0 && g <= 255 && b >= 0 && b <= 255) {
+            color = QColor(r, g, b);
+        }
+        lua_pop(L, nResult);
+        return mapInfoProperties{ text, isBold, isItalic, color };
+    });
+    host.mpMap->mpMapper->updateInfoContributors();
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#killMapInfoProvider
+int TLuaInterpreter::killMapInfoProvider(lua_State* L)
+{
+    auto& host = getHostFromLua(L);
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "killMapInfoProvider: bad argument #1 type (label as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    auto name = lua_tostring(L, 1);
+    
+    host.mpMap->mpMapper->mMapInfoContributorManager->contributors.remove(name);
+    host.mpMap->mpMapper->updateInfoContributors();
+    host.mMapInfoProviders.remove(name);
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#toggleMapInfoProvider
+int TLuaInterpreter::toggleMapInfoProvider(lua_State* L)
```

Comment:
For consistency better to use `enable` and `disable` with other functions. For non-programmers toggles `true` and `false` don't make a huge amount of sense straight away